updated post-vote release instructions (#304)
apache/openwhisk-release: docs/release_instructions.md

`config.json` file by incrementing the `rc` number and changing git
hashes.

### Publishing a Successful Release to Apache Dist Servers

After a successful vote, the release manager will commit the artifacts

> being released to the openwhisk subdir of the Apache dist svn.

Reviewer: openwhisk
```suggestion
being released to the OpenWhisk subdir of the Apache dist svn.
```

> The upload_to_dist.sh script automates the copy & svn add operations, but it

Reviewer: ```suggestion
The [`upload_to_dist.sh`](../tools/upload_to_dist.sh) script automates the copy & svn add operations, but it
```

i have to look up the path.

assumes that every file found in the release candidate subdir should be released.

> If this is not true (eg multiple parallel release votes), then the upload

Reviewer: ```suggestion
If this is not true (e.g., multiple parallel release votes), then the upload
```

must be performed manually.
```
./upload_to_dist.sh ../release-configs/<MY_RELEASE_CONFIG>.json
```

Assuming the expected set of files were added, commit them:
```
cd ../stagingArea/svn_release && svn commit -m  "Apache OpenWhisk X.Y.Z release of <Component Description>"
```

Relatively soon after doing the svn commit, you should receive an email from
from reporter.apache.org asking you to add your version data
to its database.  Please follow the link and add the information

> (the data is used to generate periodic reports to the ASF Board).

Reviewer: does it make sense to include a sample email?

tags (`git tag`) and then add a new tag following the same convention using the
git commit hash from <MY_RELEASE_CONFIG>.json.  After tagging a repo, push the tag.

Many of the GitHub repositories are configured to build binary artifacts in response
to new tags being committed.  Monitor the build process and ensure that all expected
artifacts are created for each tag you commit.

There are some slightly outdated, but much more detailed comments
on [release tagging](tag_release.md) available if you need a reminder
of the git commands to use.

### Create GitHub releases

After pushing the tags, you should go to the GitHub Releases for
each released project and "Draft a new release" using the tag you just
pushed. If the project contains a CHANGELOG or RELEASENOTES, copy that
information into the release description.

### Dockerhub updates

If the components you released build docker images, then you should
build the docker images locally, tag them with the release version
(following the naming scheme for the repo), push the new images to
dockerhub using the whiskbot dockerhub id, and update the `latest`
tag to point to the new images.

> If you have published new images to dockerhub, submit PRs to openwhisk-deploy-kube

Reviewer: ```suggestion
If you have published new images to dockerhub, submit PRs to [openwhisk-deploy-kube](https://github.com/apache/openwhisk-deploy-kube)
```

> and openwhisk-devtools (docker-compose) to use the new images.

Reviewer: ```suggestion
and [openwhisk-devtools (docker-compose)](https://github.com/apache/openwhisk-devtools) to use the new images.
```

### Update Downloads Page

> Submit a PR to openwhisk-website to update the Downloads page to

Reviewer: ```suggestion
Submit a PR to [openwhisk-website](https://github.com/apache/openwhisk-website) to update the Downloads page to
```

tag to point to the new images.

If you have published new images to dockerhub, submit PRs to
openwhisk-deploy-kube and openwhisk-devtools (docker-compose)
to use the new images.

### Homebrew

If you released a new version of openwhisk-cli, then submit a PR
to Homebrew to publish the new cli version.

### Update Downloads Page

Submit a PR to openwhisk-website to update the Downloads page to
refer to the newly released versions.

### Announcing the Release

For all normal releases you should wait at least 24 hours before
announcing the release to allow time for Apache Dist mirrors to
be updated with the newly released artifacts.  You must also wait
until the PR to update the website has been merged and the
change appears on the website (successful Jenkins job to rebuild website).

When announcing a release, you must use the URL of the OpenWhisk Downloads
page (or some other URL that supports mirroring).  Do not include a direct
link to the dist.apache.org svn server.

Releases should always be announced to dev@openwhisk.apache.org.
Releases can optionally be announced to announce@apache.org at the discretion
of the release manager.

### Post-release cleanup

> 1. Remove the release candidate files from the staging svn

Reviewer: ```suggestion
1. Remove the release candidate files from the staging svn.
```
